features/fixtures/shared/scenarios/Scenario.m

```diff
@@ -45,10 +45,36 @@ - (instancetype)initWithConfig:(BugsnagConfiguration *)config {
     return self;
 }
 
+- (void)waitForNetworkConnectivity {
+    NSDictionary *proxySettings = (__bridge_transfer NSDictionary *)CFNetworkCopySystemProxySettings();
+    NSLog(@"*** Proxy settings = %@", proxySettings);
+    
+    // BrowserStack appears to use a proxy to establish connectivity with Maze Runner...
+    // bs-local.com resolves to 127.0.0.1 and there is nothing listening on port 9339 locally.
+    
+    NSURL *url = [NSURL URLWithString:self.config.endpoints.notify];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:3];
+    while (1) {
+        NSDate *retryDate = [NSDate dateWithTimeIntervalSinceNow:1];
+        NSURLResponse *response = nil;
+        NSError *error = nil;
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:&error];
+#pragma clang diagnostic pop
+        NSLog(@"*** %@", response ?: error);
+        if ([response isKindOfClass:[NSHTTPURLResponse class]] && ((NSHTTPURLResponse *)response).statusCode / 100 == 2) {
+            break;
+        }
```

Comment:
Could we consider putting an `else` on this to log something like "MazeRunner probe failed, retrying in 1 second"?  We could then gain easier insight into how frequently it actually helps us out.

Reply:
Done 👍